R/utilities-simulation.R

```diff
@@ -244,17 +236,94 @@ createSimulationBatch <- function(simulation, parametersOrPaths = NULL, molecule
     variableMolecules <- unlist(lapply(variableMolecules, function(x) x$path))
   }
 
-  simulationBatchFactory <- getNetTask("SimulationBatchFactory")
-
   simulationBatchOptions <- SimulationBatchOptions$new(
     variableParameters = variableParameters,
     variableMolecules = variableMolecules
   )
 
-  net <- rClr::clrCall(simulationBatchFactory, "Create", simulation$ref, simulationBatchOptions$ref)
+  net <- rClr::clrNew("OSPSuite.R.Services.SettingsForConcurrentRunSimulationBatch", simulation$ref, simulationBatchOptions$ref)
   SimulationBatch$new(net, simulation)
 }
 
+#' Run simulation batches
+#' @details Runs a set of simulation batches. The simulation batches must be populated
+#' with sets of parameter and start values with \code{SimulationBatch$addRunValues()}
+#' prior to running. After the run, the list of parameter and start values is cleared.
+#'
+#' @param simulationBatches List of \code{SimulationBatch} objects with added parameter and initial values
+#' @param simulationRunOptions Optional instance of a \code{SimulationRunOptions} used during the simulation run.
+#' @param silentMode If \code{TRUE}, no warnings are displayed if a simulation fails.
+#' Default is \code{FALSE}.
+#'
+#' @return Nested list of \code{SimulationResults} objects. The first level of the list are the IDs of the simulations of SimulationBatches, containing a list of \code{SimulationResults} for each set of parameter/initial values. If a simulation with a parameter/initial values set fails, the result for this run is \code{NULL}
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' sim <- loadSimulation("simple", loadFromCache = TRUE)
+#' parameters <- c("Organism|Liver|Volume", "R1|k1")
+#' molecules <- "Organism|Liver|A"
+#' # Create two simulation batches. In this case, they are based on the same simulation
+#' # and vary the same paramter and initial values.
+#' simulationBatch1 <- createSimulationBatch(simulation = sim,
+#' parametersOrPaths = parameters,
+#' moleculesOrPaths = molecules)
+#' simulationBatch2 <- createSimulationBatch(simulation = sim,
+#' parametersOrPaths = parameters,
+#' moleculesOrPaths = molecules)
+#' #Ids of run values
+#' ids <- c()
+#' ids[[1]] <- simulationBatch1$addRunValues(parameterValues = c(1, 2), initialValues = 1)
+#' ids[[2]] <- simulationBatch1$addRunValues(parameterValues = c(1.6, 2.4), initialValues = 3)
+#' ids[[3]] <- simulationBatch2$addRunValues(parameterValues = c(4, 2), initialValues = 4)
+#' ids[[4]] <- simulationBatch2$addRunValues(parameterValues = c(2.6, 4.4), initialValues = 5)
+#' res <- runSimulationBatches(simulationBatches = list(simulationBatch1, simulationBatch2))

```

Comment:
can it be an array e.g. c(simulationBatch1, simulationBatch2)

Reply:
`c()` is always a list for non-atomic types. I prefer to use `list` in the code when I know it will be casted to a list.

I wish R were strong typed...